pkg/postage/postagecontract/contract.go

```diff
@@ -131,6 +131,11 @@ func (c *postageContract) expiredBatchesExists(ctx context.Context) (bool, error
 	if err != nil {
 		return false, err
 	}
+
+	if len(results) == 0 {
```

Comment:
I don't know if it makes sense, but since we expect exactly one result - we could just check if `len != 1` and specify in the error how many results we received?